source/common/router/upstream_request.cc

```diff
@@ -205,11 +205,9 @@ void UpstreamRequest::cleanUp() {
     }
   }
 
+  stream_info_.setStreamState(StreamInfo::StreamState::Ended);
```

Comment:
I'm confused why we need this.
One can already infer most of this from various timing characteristics and I think having a separate round of state we need to keep track of is problematic
I see this isn't technically a part of this PR but landed in https://github.com/envoyproxy/envoy/pull/26256 but that somehow was merged without senior maintainer sign-off and I think we may need to revisit it and wait on this one until that's sorted out.

Reply:
The follow PR: https://github.com/envoyproxy/envoy/pull/26203 change the behavior of %DURATION% so it's now returning a value at any time queried during a stream. When periodic access logs were originally added, Matt had a concern about not having a way to distinguish between an end of stream access log and mid-stream access log. Therefore, instead of %DURATION% (that was null for mid-stream access logs), %STEAM_STATE% was introduced, which returns either Started, InProgress or Ended. The streamState() is currently only being used for this purpose. Please let me know if this makes more sense now